### PR TITLE
Migrate to slalpha4

### PR DESCRIPTION
**Pull request by Chuhaa**
## Purpose

- Migrate to slalpha4
- Add display annotation

## Goals

- Fix https://github.com/wso2-enterprise/choreo/issues/3410
- Fix https://github.com/wso2-enterprise/choreo/issues/2963

## Automation tests
 - Unit tests 
    Covered all the client operations.
 - Integration tests
    Covered all the client operations.

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Test environment

- Ballerina SLAlpha4
- Ubuntu 20.04 LTS

### CODE REVIEW

File: client.bal [hunk 1]
```diff
@@ -73,13 +73,15 @@ public client class Client {
     # + cannedACL - The access control list of the new bucket.
     # 
     # + return - If failed turns error.
-    remote function createBucket(string bucketName, CannedACL? cannedACL = ()) returns @tainted error? {
-        map<string> requestHeaders = {};
+    @display {label: "Create bucket"}
+    remote function createBucket(@display {label: "Bucket name"} string bucketName,
+                                    @display {label: "Grant"} CannedACL? cannedACL = ()) returns @tainted error? {
```
Comment (abeykoon) [inline client.bal:78]: Shall we update display name Grant --> Access Control List

File: client.bal [hunk 2]
```diff
@@ -46,23 +47,22 @@ public client class Client {
...
     # 
     # + return - If success, returns a list of Bucket record, else returns error
+    @display {label: "Get buckets"}
     remote function listBuckets() returns @tainted Bucket[]|error {
-        map<string> requestHeaders = {};
-        http:Request request = new;
-
-        requestHeaders[HOST] = self.amazonHost;
-        requestHeaders[X_AMZ_CONTENT_SHA256] = UNSIGNED_PAYLOAD;
+        map<string> requestHeaders = {
```
Comment (abeykoon) [inline client.bal:52]: As this code section is repeated, shall we move to a separate function and call?

File: client.bal [hunk 3]
```diff
@@ -115,59 +117,68 @@ public client class Client {
...
     # 
     # + return - If success, returns S3Object[] object, else returns error
-    remote function listObjects(string bucketName, string? delimiter = (), string? encodingType = (), int? maxKeys = (),
-                                string? prefix = (), string? startAfter = (), boolean? fetchOwner = (), 
-                                string? continuationToken = ()) returns @tainted S3Object[]|error {
-        map<string> requestHeaders = {};
-        map<string> queryParamsMap = {};  
-        http:Request request = new;
+    @display {label: "Get objects"}
+    remote function listObjects(@display {label: "Bucket name"} string bucketName,
```
Comment (abeykoon) [inline client.bal:121]: If there is a large file in the bucket, how we are going to handle it?

File: client.bal (hunk 2 above)
Comment (Chuhaa) [reply client.bal]: Fixed in https://github.com/ballerina-platform/module-ballerinax-aws.s3/commit/02312d664879220bb3bfe2db0e8786ae7af97a06

File: client.bal (hunk 3 above)
Comment (Chuhaa) [reply client.bal:115]: We need to handle this in future improvements. Created an issue to track. https://github.com/wso2-enterprise/choreo/issues/3531

**Review by abeykoon — APPROVED** [review]

**Pull request merged**